LdapForNet/LdapEntry.cs

```diff
@@ -215,33 +215,72 @@ public class DirectoryModificationAttribute : DirectoryAttribute
             Native.Native.LdapModOperation.LDAP_MOD_REPLACE;
     }
 
-    public class SearchResultAttributeCollection : KeyedCollection<string, DirectoryAttribute>
+    public abstract class DirectoryAttributeCollectionBase<T> : List<T>
+        where T : DirectoryAttribute
     {
-        internal SearchResultAttributeCollection()
-            : base(StringComparer.OrdinalIgnoreCase)
+        public IEnumerable<string> AttributeNames =>
+            this.Select(x => x.Name);
+        
+        public bool Contains(string attribute) =>
+            this.Any(x => String.Equals(x.Name, attribute, StringComparison.OrdinalIgnoreCase));
+
+        public DirectoryAttribute this[string attribute]
         {
+            get
+            {
+                var item = this.FirstOrDefault(x => String.Equals(x.Name, attribute, StringComparison.OrdinalIgnoreCase));
+
+                if (item == null)
+                {
+                    throw new KeyNotFoundException();
+                }
+
+                return item;
+            }
         }
 
-        public ICollection<string> AttributeNames => Dictionary.Keys;
+        public bool TryGetValue(string attribute, out DirectoryAttribute item)
+        {
+            item = this.FirstOrDefault(x => String.Equals(x.Name, attribute, StringComparison.OrdinalIgnoreCase));
+
+            if (item == null)
+            {
+                return false;
+            }
+
+            return true;
+        }
 
-        protected override string GetKeyForItem(DirectoryAttribute item)
+        public bool Remove(string attribute)
         {
-            return item.Name;
+            var found = false;
+
+            for (int i = 0; i < Count; i++)
+            {
+                if (String.Equals(this[i].Name, attribute, StringComparison.OrdinalIgnoreCase))
+                {
+                    RemoveAt(i);
+                    --i;
+
+                    found = true;
+                }
+            }
+
+            return found;
         }
     }
 
-    public class ModifyAttributeCollection : KeyedCollection<string, DirectoryModificationAttribute>
+    public class SearchResultAttributeCollection : DirectoryAttributeCollectionBase<DirectoryAttribute>
```

Comment:
SearchResultAttributeCollection should remain a keyedCollection for quick lookup when querying LDAP with many records.

Reply:
Yes. It's not good for performance, but I think no more 200-300 attributes records.